packages/docusaurus-theme-classic/src/theme/SiteMetadata/index.tsx

```diff
@@ -24,10 +24,18 @@ import SearchMetadata from '@theme/SearchMetadata';
 // See https://github.com/facebook/docusaurus/issues/3317
 function AlternateLangHeaders(): JSX.Element {
   const {
-    i18n: {defaultLocale, localeConfigs},
+    i18n: {currentLocale, defaultLocale, localeConfigs},
   } = useDocusaurusContext();
   const alternatePageUtils = useAlternatePageUtils();
 
+  const currentHtmlLang = localeConfigs[currentLocale]?.htmlLang;
```

Comment:
```suggestion
  const currentHtmlLang = localeConfigs[currentLocale]!.htmlLang;
```

We can assume each locale will have a config in the map